core/shape-parts.cpp

```diff
@@ -39,6 +39,32 @@ const Path& BasicBezier::generate(Point endPoint, Point transitionPoint) {
   return m_path;
 }
 
+const Path& FlatBezier::generate(Point endPoint, Point transitionPoint) {
+  MCFixed flatEndY = m_flatWidth / -2; // WHY DO I HAVE TO MAKE THIS NEGATIVE TO GET IT TO WORK???
```

Comment:
So I think I've finally worked this out. Monkeycam actually draws the board outline entirely in the negative Y-space then mirrors it. All of the transition and edge points have negative Y initially. This explains why the flat width has to be negative as well. The key section of the code where this happens is in [shapes.cpp lines 112-118](https://github.com/mikemag/MonkeyCAM/blob/ded90afaed3f29eff5f6490dac6081fb2608de5d/core/shapes.cpp#L112-L118). I presume you set it up this way because you want the pats to be counter-clockwise for a specific reason?

With this in mind I think I need to leave the negative sign in (or make some significant changes elsewhere in the code!?)